### PR TITLE
Support sequences without using maximum(:id)

### PR DESCRIPTION
**Pull request by davidcelis**
Minifacture is unable to run sequences for models that don't use an
integer for their primary keys (e.g. UUIDs). To solve this, we can have
each factory maintain a count that is incremented as `build` is called.
This count is also inherited from parents to simulate how sequences
would work in cases of single table inheritance.

Signed-off-by: David Celis me@davidcel.is

### CODE REVIEW

**Comment by stephencelis** [comment]
Hm. What happens if you have a model with a `count` property?

**Comment by davidcelis** [comment]
Yeah I suspected that may not work; it looks like the `count` property gets set to `nil`. because of `h.except(:count)`. I'm not seeing a good place to store this that doesn't get mucked in with the attributes. Any thoughts?

**Comment by davidcelis** [comment]
switched to adding a class variable to maintain counts, hope u like

File: minifacture.rb [hunk 1]
```diff
@@ -13,22 +13,22 @@
 #   end
 class Minifacture < Struct.new(:__klass__)
   undef_method *instance_methods.grep(/^(?!__|object_id)/)
-  @@attrs = {} and private_class_method :new
+  @@attrs = {} and @@counts = {} and private_class_method :new
```
Comment (stephencelis) [inline minifacture.rb]: Does `@@counts = Hash.new 0` work and save us the line below?

File: minifacture_test.rb [hunk 2]
```diff
@@ -1,11 +1,11 @@
 require './minifacture'
-require 'test/unit'
+require 'minitest/autorun'
```
Comment (stephencelis) [inline minifacture_test.rb:2]: :fireworks:

File: minifacture.rb (hunk 1 above)
Comment (davidcelis) [inline minifacture.rb]: Yup. Updated

**Comment by stephencelis** [comment]
Thanks!

**Pull request merged**